gui-agent/vmside.c

```diff
@@ -1505,54 +1504,74 @@ static void mkghandles(Ghandles * g)
                 g->screen),
             WhitePixel(g->display,
                 g->screen));
+    if ((unsigned)snprintf(tray_sel_atom_name, sizeof(tray_sel_atom_name),
+            "_NET_SYSTEM_TRAY_S%u", DefaultScreen(g->display)) >=
+        sizeof tray_sel_atom_name)
+        abort();
+#define SUPPORTED_ATOMS 6
+    const struct {
+        Atom *const dest;
+        const char *const name;
+    } atoms_to_intern[] = {
+        { &g->wmDeleteMessage,  "WM_DELETE_MESSAGE" },
+        { &g->wmProtocols,      "WM_PROTOCOLS" },
+        { &g->wm_hints,         "WM_HINTS" },
+        { &g->wm_class,         "WM_CLASS" },
+        { &g->tray_selection,   tray_sel_atom_name },
+        { &g->tray_opcode,      "_NET_SYSTEM_TRAY_MESSAGE_OPCODE" },
+        { &g->xembed_info,      "_XEMBED_INFO" },
+        { &g->utf8_string_atom, "UTF8_STRING" },
+        { &g->wm_state,         "WM_STATE" },
+        { &g->net_wm_state,     "_NET_WM_STATE" },
+        { &g->wm_state_fullscreen, "_NET_WM_STATE_FULLSCREEN" },
+        { &g->wm_state_demands_attention, "_NET_WM_STATE_DEMANDS_ATTENTION" },
+        { &g->wm_take_focus,    "WM_TAKE_FOCUS" },
+        { &g->net_wm_name,      "_NET_WM_NAME" },
+        { &g->wm_normal_hints,  "WM_NORMAL_HINTS" },
+        { &g->clipboard,        "CLIPBOARD" },
+        { &g->targets,          "TARGETS" },
+        { &g->qprop,            "QUBES_SELECTION" },
+        { &g->compound_text,    "COMPOUND_TEXT" },
+        { &g->xembed,           "_XEMBED" },
+    };
+    Atom supported[SUPPORTED_ATOMS + QUBES_ARRAY_SIZE(atoms_to_intern)];
     /* pretend that GUI agent is window manager */
-    g->net_wm_name = XInternAtom(g->display, "_NET_WM_NAME", False);
-    net_supporting_wm_check = XInternAtom(g->display, "_NET_SUPPORTING_WM_CHECK", False);
-    net_supported = XInternAtom(g->display, "_NET_SUPPORTED", False);
-    supported[0] = net_supported;
-    supported[1] = net_supporting_wm_check;
-    /* _NET_WM_MOVERESIZE required to disable broken GTK+ move/resize fallback */
-    supported[2] = XInternAtom(g->display, "_NET_WM_MOVERESIZE", False);
-    supported[3] = XInternAtom(g->display, "_NET_WM_STATE", False);
-    supported[4] = XInternAtom(g->display, "_NET_WM_STATE_FULLSCREEN", False);
-    supported[5] = XInternAtom(g->display, "_NET_WM_STATE_DEMANDS_ATTENTION", False);
+    const char *names[SUPPORTED_ATOMS + QUBES_ARRAY_SIZE(atoms_to_intern)] = {
+        "_NET_SUPPORTED",
+        "_NET_SUPPORTING_WM_CHECK",
+        /* _NET_WM_MOVERESIZE required to disable broken GTK+ move/resize fallback */
+        "_NET_WM_MOVERESIZE",
+        "_NET_WM_STATE",
+        "_NET_WM_STATE_FULLSCREEN",
+        "_NET_WM_STATE_DEMANDS_ATTENTION",
+    };
+    for (size_t i = 0; i < QUBES_ARRAY_SIZE(atoms_to_intern); ++i)
+        names[SUPPORTED_ATOMS + i] = atoms_to_intern[i].name;
+    if (!XInternAtoms(g->display, (char **)names,
+                      QUBES_ARRAY_SIZE(atoms_to_intern), False, supported)) {
```

Comment:
`QUBES_ARRAY_SIZE(names)`

Reply:
Fixed in 59ddfd3
I've manually verified that nm-applet works now.
At this point, it's pretty obvious we don't have tray icon tests in our test suite...

Reply:
Yes, now it is ok.